frontend/src/reducers/exercise/index.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+import { createSelector } from 'reselect';
+
+const INITIAL_STATE = {
+  timer: 0, // exercise time elapsed in seconds
+};
+
+export default function reducer(state = INITIAL_STATE, action = {}) {
+  switch (action.type) {
+    case 'exercise/TIMER_TICK': {
+      return {
+        ...state,
+        timer: (state.timer += action.payload.amount),
+      };
+    }
+
+    default: {
+      return state;
+    }
+  }
+}
+
+// Actions
+export function tickTimer(amount = 10) {
+  return {
+    type: 'exercise/TIMER_TICK',
+    payload: {
+      amount,
+    },
+  };
+}
+
+// Selectors
+const exerciseSelector = state => state.exercise;
+
+export const getExerciseTimer = createSelector(
+  exerciseSelector,
+  exercise => exercise.timer
+);
```

Comment:
what does this selector do?
is it the same as getting state.exercise.timer ?

Reply:
Yep it is the same as getting `state.exercise.timer` but it is memoized by reselect.